### DISCUSSION TITLE
Extreme Overclocking

### DISCUSSION THREAD

**Discussion by PaintYourDragon** [earlephilhower/arduino-pico#1179] [Ideas]
I’m aware how absurd this sounds, but there’s an actual use case, so bear with me…

[PicoDVI](https://github.com/Wren6991/PicoDVI) generates DVI-compatible signals from an overclocked RP2040. 252 MHz for the lowest resolution (640x480) and going up from there.

PicoDVI is normally a Pico-SDK project. I’m working on an Arduino wrapper to make this easier to use. The lower resolutions are working fine in this setting now (640x480p, 252 MHz CPU overclock, and 800x480p, 295.2 MHz), but other resolutions (all using higher clock rates) do not…but they _do_ all work in the original Pico-SDK build of essentially the same code.

After some fussing around, I found that the Arduino core just doesn’t work at the same clock rates. This is even with just the basic blink sketch. At first I thought this was some 8-bit or integer overflow-related thing, because 256 MHz works but 257 does not…but no, that’s not it, because 295.2 MHz is fine. 275 and 300 MHz (from the tools menu) also lock up, but 295.2 (added manually to menu) does not. Tested across multiple boards.

Before I go digging deeper…since you’re closer to this code, I was wondering if you might know a likely culprit. The clock-related code goes quickly over my head.

Thanks for any insights.

**Comment by earlephilhower** [comment]
I haven't really looked into it, but my *guess* is the SPI <->flash interface is being overclocked (since it is a division of another internal clock.

Try using the `Generic` board and the `Boot->Stage 2->Generic SPI / 4` to use the slowest, most compatible interface.

It could also be a function if the code being in RAM and not flash in the PicoDVI.  There are pico-sdk macros you can use to manually push functions into RAM from flash, but the saner thing would be to rewrite the linker file to place everything in RAM.

Also, will move to discussion since it's not really a core issue here...
